### PR TITLE
Moonlight build / deploy

### PR DESCRIPTION
**Pull request by packplusplus**

### CODE REVIEW

File: swift/probe/README.md [hunk 1]
```diff
@@ -1,15 +1,45 @@
 # Moonlight
 Probe written in [swift](https://www.swift.org/)
 
-## Getting started
+## Mac Getting started
 
-Compile Moonlight, replacing ```<TARGET>``` with a supported operating system:
+Assuming you are on an ARM mac.
```
Comment (privateducky) [inline swift/probe/README.md:6]: this README needs to be clearer on the steps to install this probe on each system: Linux and MacOS right now. 

we should organize it as:

- what is this probe: description
- how to download a precompiled probe via our API
- how to install this probe from source from a Mac ARM, Mac intel, Linux intel (separate sections for each)

File: swift/probe/build.bash [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# run with bash -e on an m1 mac
+swiftc -Osize ./Sources/Moonlight/moonlight.swift -o moonlight_darwin-x86_64 -target x86_64-apple-macos10.13 && strip moonlight_darwin-x86_64
+
+swiftc -Osize ./Sources/Moonlight/moonlight.swift -o moonlight_darwin-arm64 -target arm64-apple-macos10.13 && strip moonlight_darwin-arm64
```
Comment (privateducky) [inline swift/probe/build.bash:4]: I don't like this file, as it's a specific install script from a specific POV (arm/Mac). see readme comment on making the process described clearer from each perspective.

**Review by privateducky — CHANGES_REQUESTED** [review]
we need clearer instructions

File: swift/probe/README.md (hunk 1 above)
Comment (packplusplus) [reply swift/probe/README.md:6]: Completely fair. Updated

File: swift/probe/build.bash (hunk 2 above)
Comment (packplusplus) [reply swift/probe/build.bash]: Removed

**Comment by packplusplus** [comment]
@privateducky can you re-review the instructions?

File: swift/probe/README.md [hunk 3]
```diff
@@ -1,15 +1,175 @@
 # Moonlight
-Probe written in [swift](https://www.swift.org/)
+A multiplatform etect probe written in [swift](https://www.swift.org/).
```
Comment (makk94) [inline swift/probe/README.md:2]: etect -> Detect

File: swift/probe/README.md [hunk 4]
```diff
@@ -1,15 +1,175 @@
...
+```
+export PRELUDE_TOKEN='<INSERT TOKEN>'
+curl -X GET -L "https://detect.prelude.org/download/moonlight?dos=darwin-arm64" -H "token:${PRELUDE_TOKEN}" > moonlight
+chmod +x moonlight
+./moonlight
+```
+
+### Intel Mac Download
+
+Intel based Mac's are "x86_64" processors. A majority of usable macs bought before 2020 are Intel Bsed.
```
Comment (makk94) [inline swift/probe/README.md:23]: based

File: swift/probe/README.md [hunk 5]
```diff
@@ -1,15 +1,175 @@
 # Moonlight
-Probe written in [swift](https://www.swift.org/)
+A multiplatform etect probe written in [swift](https://www.swift.org/).
 
-## Getting started
+## Downloading Precompiled Binaries
```
Comment (makk94) [inline swift/probe/README.md:4]: imo redirect this section entirely to docs.prelude.org so we only need to update one spot for future changes

File: .github/workflows/moonlight-dev.yaml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    steps:
+      - name: Check out code. 
+        uses: actions/checkout@v3
+
+      - name: Build Darwin x86
+        working-directory: swift/probe
+        run: |
+          swiftc -Osize ./Sources/Moonlight/moonlight.swift -o moonlight_darwin-x86_64 -target ${{ env.MAC_TARGET }}
+          strip moonlight_darwin-x86_64
+          ls -l $PWD
```
Comment (makk94) [inline .github/workflows/moonlight-dev.yaml:36]: whats this for?

File: swift/probe/README.md [hunk 7]
```diff
@@ -1,15 +1,175 @@
...
+  ```
+  swiftc -Osize ./Sources/Moonlight/moonlight.swift -o moonlight_darwin-arm64 -target arm64-apple-macos10.13
+  strip moonlight_darwin-arm64
+  ```
+* Export PRELUDE_TOKEN `export PRELUDE_TOKEN=<SECRET>`
+* Next, test an executable: 
+  * x86 mac: `./moonlight_darwin-x86_64`
+  * m1 or m2 mac: `./moonlight_darwin-arm64`
+
+### Intel Mac Compiling
```
Comment (makk94) [inline swift/probe/README.md:54]: everything in this section seems exactly the same as the previous...

File: .github/workflows/moonlight-dev.yaml [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+          aws-access-key-id: ${{ secrets.ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.SECRET_ACCESS_KEY }}
+
+      - name: Copy files to the s3 with the AWS CLI
+        working-directory: swift/probe
+        run: |
+          aws s3 cp moonlight_darwin-x86_64 s3://${{ env.S3_BUCKET }}/prelude/probes/moonlight/
+          aws s3 cp moonlight_darwin-arm64 s3://${{ env.S3_BUCKET }}/prelude/probes/moonlight/
+  
+  # build_linux:
```
Comment (mireaulf) [inline .github/workflows/moonlight-dev.yaml:57]: Remove if we don't plan on using it

**Pull request closed without merging**